File: Week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int count_letters (string input);
int count_words (string input);
int count_sentences (string input);

int main(void)
{
    string text = get_string("Text: ");

    float L = ((float) count_letters(text) / (float) count_words(text)) * 100;
    float S = ((float) count_sentences(text) / (float) count_words(text)) * 100;

    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    int final = round(index);

    if (final > 16)
    {
        printf("Grade 16+\n");
    }
    else if (final < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n" , final);
    }
}

int count_letters (string input)
{
    int output = 0;
    int letter = 0;

    for (int i = 0; input[i] != '\0'; i++)
    {
        if (input[i] >= 'a' && input[i] <= 'z')
        {
            letter = 1;
        }
        else if (input[i] >= 'A' && input[i] <= 'Z')
        {
            letter = 1;
        }
        else
        {
            letter = 0;
        }   

        output += letter;
    }

    return output;
}

int count_words (string input)
{
    int output = 0;
    int letter = 0;
    int word = 0;

    for (int i = 0; input[i] != '\0'; i++)
    {
        if (input[i] >= 'a' && input[i] <= 'z')
        {
            letter = 1;
        }
        else if (input[i] >= 'A' && input[i] <= 'Z')
        {
            letter = 1;
        }
        else
        {
            letter = 0;
        }

        if (letter == 1 && input[i+1] == ' ')
        {
            word = 1;
        }
        else if (letter == 1 && input[i+1] == '.')
        {
            word = 1;
        }
        else if (letter == 1 && input[i+1] == '!')
        {
            word = 1;
        }
        else if (letter == 1 && input[i+1] == '?')
        {
            word = 1;
        }
        else if (letter == 1 && input[i+1] == ',')
        {
            word = 1;
        }
        else
        {
            word = 0;
        }

        output += word;

    }

    return output;
}

int count_sentences (string input)
{
    int total_sentence = 0;

    for (int i = 0; input[i] != '\0'; i++)
    {
        int sentence = 0;

        if (input[i] == '.' || input[i] == '!' || input[i] == '?')
        {
            sentence = 1;
        }
        else
        {
            sentence = 0;
        }

        total_sentence += sentence;
    }

    return total_sentence;
}

File: Week6/sentimental-readability/readability.py
# TODO
from cs50 import get_string
import math

def main():
    text = get_string("Text: ")

    l = avg_letters(text)
    s = avg_sentences(text)
    score = 0.0588 * l - 0.296 * s - 15.8

    if score > 16:
        print("Grade 16+")
    elif score < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(score)}")


def count_words(text):
    word_count = 0
    for i in range(len(text)):
        if text[i].isalpha() and text[i + 1] == " ":
            word_count += 1
        elif text[i] == "." or text[i] == "?" or text[i] == "!" or text[i] == ",":
            if text[i -1].isalpha:
                word_count += 1
    return word_count


def avg_letters(text):
    char_count = 0
    for i in range(len(text)):
        if text[i].isalpha():
            char_count += 1
    avg = (char_count / count_words(text)) * 100
    return avg


def avg_sentences(text):
    sentence_count = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "?" or text[i] == "!":
            sentence_count += 1
    avg = (sentence_count / count_words(text)) * 100
    return avg


main()